File: WebsocketDealer/struct.go
package WebsocketDealer

import (
	"reflect"
	"sync"

	"github.com/fasthttp/websocket"
	jsoniter "github.com/json-iterator/go"
)

var sessionType = reflect.TypeOf((*Session)(nil))
var replierType = reflect.TypeOf((*Replier)(nil))

type StandardCall struct {
	Function string          `json:"function"`
	Argument []interface{}   `json:"argument"`
	Id       jsoniter.Number `json:"id"`
	IsReply  bool            `json:"reply"` //只是保留，客户端不需要实现 若为Reply时，需要设置为true
}
type StandardReply struct {
	Id   jsoniter.Number `json:"id"`
	Data interface{}     `json:"data"`
}
type Replier struct { //回复者
	conn              *websocket.Conn
	idCounter         *uint64
	id                jsoniter.Number
	bindReplyId       func(id uint64, c chan StandardReply)
	removeBindReplyId func(id uint64)
	haveReplied       bool //是否已经进行了回复。如果没有进行回复，RWeb将自动回复nil
	coder             Coder
}

var replierPool = &sync.Pool{
	New: func() interface{} {
		return new(Replier)
	},
}

type WebsocketResponse struct {
	Id      jsoniter.Number `json:"id"`
	Data    interface{}     `json:"data"`
	IsReply bool            `json:"reply"`
}

var responsePool = &sync.Pool{
	New: func() interface{} {
		return new(WebsocketResponse)
	},
}
var requestPool = &sync.Pool{
	New: func() interface{} {
		return new(StandardCall)
	},
}
var Once sync.Once

func InitWebsocketDealer() {

}

File: clientDart/struct.dart
class StandardCall {
  late String Function;
  late List<dynamic> Argument;
  late int Id;
  late bool IsReply;
  StandardCall(
      {required this.Function,
      required this.Argument,
      required this.Id,
      required this.IsReply}) {}
  StandardCall.fromJson(Map<String, dynamic> json) {
    Function = json["function"];
    Argument = json["argument"];
    Id = json["id"];
    IsReply = json["reply"];
  }
  Map<String, dynamic> toJson() =>
      {'function': Function, 'argument': Argument, "id": Id, "reply": IsReply};
}

class StandardReply {
  late int Id;
  late bool Reply;
  late dynamic Data;
  StandardReply({required this.Id, required this.Reply, required this.Data}) {}
  Map<String, dynamic> toJson() {
    return {"id": Id, "reply": Reply, "data": Data};
  }

  StandardReply.fromJson(Map<String, dynamic> json) {
    Id = json["id"];
    Data = json["data"];
    Reply = json["reply"];
  }
}
/*
type StandardReply struct {
	Id    jsoniter.Number `json:"id"`
	Data  interface{}     `json:"data"`
	Reply bool            `json:"reply"`
}
*/
